internal/gps/vcs_source_test.go

```diff
@@ -524,6 +524,62 @@ func testHgSourceInteractions(t *testing.T) {
 	<-donech
 }
 
+func Test_gitSource_listVersions_noHEAD(t *testing.T) {
+	t.Parallel()
+
+	requiresBins(t, "git")
+
+	h := test.NewHelper(t)
+	defer h.Cleanup()
+	h.TempDir("smcache")
+	cpath := h.Path("smcache")
+	h.TempDir("repo")
+	repoPath := h.Path("repo")
+
+	// Create test repo with a single commit on the master branch
+	h.RunGit(repoPath, "init")
+	h.RunGit(repoPath, "commit", "--allow-empty", `--message="Initial commit"`)
+
+	// Make HEAD point at a nonexistent branch (deleting it is not allowed)
+	// The `git ls-remote` that listVersions() calls will not return a HEAD ref
+	// because it points at a nonexistent branch
+	h.RunGit(repoPath, "symbolic-ref", "HEAD", "refs/heads/nonexistent")
+
+	un := "file://" + repoPath
+	u, err := url.Parse(un)
+	if err != nil {
+		t.Errorf("URL was bad, lolwut? errtext: %s", err)
```

Comment:
LOL! 😄
Let's make this `Fatalf`.
`t.Fatalf("Error parsing URL %s: %s", un, err)`